Source/FullScreenPokemon.ts

```diff
@@ -4888,6 +4888,55 @@ module FullScreenPokemon {
             player.FSP.killNormal(player.bordering[player.direction]);
         }
 
+        /**
+         * Makes a StrengthBoulder move.
+         *
+         * @param player   The Player.
+         * @param pokemon   The Pokemon using Strength.
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         * @todo Add an animation for what happens when the StrengthBoulder is moved.
+         */
+        partyActivateStrength(player: IPlayer, pokemon: IPokemon): void {
+            player.FSP.MenuGrapher.registerB();
+            player.FSP.MenuGrapher.registerB();
+            player.FSP.closePauseMenu();
+            player.FSP.TimeHandler.addEventInterval(
```

Comment:
This is a huge function to pass to `addEventInterval`. You can move setting `boulder`, xvel, and yvel out of it, and just call the shiftBoth inside.